client/extensions/woocommerce/state/selectors/promotions.js

```diff
@@ -1,31 +1,60 @@
 /**
  * External dependencies
  */
-import { get } from 'lodash';
+import { get, isObject } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import { getSelectedSiteWithFallback } from '../sites/selectors';
 
 export function getPromotions( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
-	const { promotions } = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId ], {} );
+	const promotions = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId, 'promotions' ], {} );
 	return promotions.promotions;
 }
 
+export function getPromotion( rootState, promotionId, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const promotions = getPromotions( rootState, siteId );
+	return promotions.find( ( p ) => promotionId === p.id ) || null;
```

Comment:
FWIW, I'm a fan of lodash's find instead - I find it easier to quickly grok, e.g.

`return find( promotions, { id: promotionId } ) || null`

Reply:
I've grown to like lodash's find lately as well. I changed the code to use it instead.